### ISSUE TITLE
Yearly budgets and reporting about yearly customer status

### ISSUE THREAD

**Issue by ChrisDeh** [kimai/kimai#4283]
### Describe the problem

Since we in our company usually maintain our revenue planning for the entire year and partly also have year-round budgets for our customers, regardless of projects, it would be super helpful with little effort to get an overview of the current status related to the current year and past years.

### Describe the solution you'd like

As a solution for a simple overview I would suggest:
- In the detail view of a customer at the boxes time budget and budget add the possibility to switch between monthly and annual evaluation.
- Add a new budget type for annual budgets.
- Create new reports for a customer evaluation, which shows you the yearly/monthly internal costs and revenues for customers.

If this feature shall be implemented, we are also happy to submit a PR for it.

**Comment by kevinpapst** [comment]
Yes, new budget types are planned, that's why there is a dropdown.
If you can supply a working PR I am happy to look at it.

Be warned upfront: the original implementation needs a partial rewrite. Currently there are many if/else regarding the monthly calculation, that obviously needs some kind of abstraction. 

So please: don't 💩 on that pile of initial code as well, but make it better 😁 s
Start with a small PR and then lets discuss next steps!
